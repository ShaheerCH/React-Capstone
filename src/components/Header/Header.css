.header {
  color: #ffff;
  padding: 20px 100px;
  display: flex;
  align-items: center;
  justify-content: space-between;
  background-color: #1d3752;
  margin-bottom: 20px;
}

.header__logo {
  font-size: 20px;
}

.header__link {
  color: #ffff;
}
