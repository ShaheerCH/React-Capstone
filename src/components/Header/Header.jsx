import React from 'react';
import { Link } from 'react-router-dom'
import './Header.css'

const Header = () => {
  return (
    <div className='header'>
      <div className="header__logo">
        <Link to='/' className='header__link'><h4>World Map</h4></Link>
      </div>

      <div className="logo__mode">
        <i className="fa-solid fa-gear"></i> Settings
      </div>
    </div>
  )
}

export default Header