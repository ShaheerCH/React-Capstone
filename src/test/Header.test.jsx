import React from 'react';
import { MemoryRouter as Router } from 'react-router-dom';
import  renderer  from 'react-test-renderer';
import { expect, it } from 'vitest';
import Header from '../components/Header/Header';

it('Renders the App Component', () => {
  const HeaderRenderer = renderer.create(
    <Router>
      <Header />
    </Router>
     ).toJSON();
  expect(HeaderRenderer).toMatchSnapshot()
})