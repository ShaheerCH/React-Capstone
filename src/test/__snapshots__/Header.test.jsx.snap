// Vitest Snapshot v1

exports[`Renders the App Component 1`] = `
<div
  className="header"
>
  <div
    className="header__logo"
  >
    <a
      className="header__link"
      href="/"
      onClick={[Function]}
    >
      <h4>
        World Map
      </h4>
    </a>
  </div>
  <div
    className="logo__mode"
  >
    <i
      className="fa-solid fa-gear"
    />
     Settings
  </div>
</div>
`;
